src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/api/dotnet/DotnetBackendHandler.scala

```diff
@@ -57,32 +58,45 @@ class DotnetBackendHandler(server: DotnetBackend)
             val t = readObjectType(dis)
             assert(t == 'c')
             val objToRemove = readString(dis)
-            JVMObjectTracker.remove(objToRemove)
+            ThreadPool.run(threadId, () => JVMObjectTracker.remove(objToRemove))
```

Comment:
I'm mostly doing this to be consistent. I can try removing the `ThreadPool` invocation here and see what happens in the tests.

Reply:
It may be safer to not run it in the ThreadPool because of the race condition above.  Ie, if the timer kicked in and ran "rmThread" and removed the `ExecutorService` from the ThreadPool map and later the destructor for the dotnet jvmobject was called and ran "rm" after, then it would spin up another `ExecutorService` for the thread that was calling "rm"  (although this threadId here may be the GC thread?)

Reply:
I took a look at `JVMObjectTracker.remove` and all it does is remove the object from the `HashMap`.  So, I agree we don't need to run this function on the `ThreadPool`.

Reply:
Reverted this line.